app/Http/Controllers/API/EventController.php

```diff
@@ -543,8 +543,9 @@ public function createEventv2(Request $request): JsonResponse
             }
         }
 
-        if ($autoapprove) {
-            Log::info("Auto-approve event $idParty");
+        // Only auto-approve if the feature is enabled AND the user has privileged role (Root, Admin, Host)
+        if ($autoapprove && $user->role <= ROLE::HOST) {
```

Comment:
This is probably right, but funky lol. I would have assumed ROOT was a higher level than HOST. But I'm assuming they are just numbers starting with 1 for highest permission.

Reply:
> But I'm assuming they are just numbers starting with 1 for highest permission.

Correct

https://github.com/iFixit/restarters/blob/1092162ced8f407567373a385799d400de278672/app/Models/Role.php#L11-L16